appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockLineChartBaseDataModel.java

```diff
@@ -0,0 +1,75 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.components.common.ComponentConstants;
+import org.pepstock.charba.client.data.Data;
+import org.pepstock.charba.client.data.DataPoint;
+import org.pepstock.charba.client.enums.CubicInterpolationMode;
+import org.pepstock.charba.client.enums.SteppedLine;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+
+/**
+ * Chart Data Model for Mock Line Chart based views.
+ */
+public abstract class MockLineChartBaseDataModel extends MockPointChartDataModel  {
+    /**
+     * Creates a new MockLineChartBaseDataModel instance.
+     * @param chartData  Data object of the Chart View.
+     */
+    public MockLineChartBaseDataModel(Data chartData) {
+        super(chartData);
+    }
+
+    @Override
+    protected void setDefaultStylingProperties() {
+        dataSeries.setBorderWidth(1);
+        dataSeries.setLineTension(0);
+        dataSeries.setShowLine(true);
+    }
+
+    @Override
+    protected void postDataImportAction() {
+        // No data points generated, fallback to default option.
+        if (dataSeries.getDataPoints().isEmpty()) {
+            setDefaultElements();
+        } else {
+            // Since we are dealing with a Scatter Data Series, sorting
```

Comment:
@ewpatton So this is a bit tricky in the library (Charba, which wraps the Chart.js Javascript library in GWT) in the sense that the "LineChartDataSeries" class only allows for continuous, 1-per-index values (think 1 index = 1 name), making it difficult to represent and do operations with real values (e.g. x=0.4) and so on. Thus, we use a Scatter Data Series in the underlying library instead, because it works like a Scatter Chart, but allows connecting points with lines, which in turn makes it more consistent with the MPAndroidChart Line Chart.

I guess the comment is a bit confusing, so I'll look into clarifying this a bit more.